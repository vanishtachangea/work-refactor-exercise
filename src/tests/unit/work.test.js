import React from 'react'
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Work from '../../components/work';

describe('Refactored Version of Work', () => {
    let component;
    it('should return result from userClient.get, when type is passed', () => {
        const component = renderer.create(
            <Work type="user" id="4" />,
        );
        let tree = component.toJSON();
        console.log(tree);
        expect(tree).toMatchSnapshot();
    })
    it('should return result from recordClient.post. when type is record', () => {
        const component = renderer.create(
            <Work type="record" />, 
        );
        let tree = component.toJSON();
        console.log(tree);
        expect(tree).toMatchSnapshot();
    })
    it('should return log when type is log and allowWrite is true', () => {
        const component = renderer.create(
            <Work type="log" allowWrite/>,
        );
        let tree = component.toJSON();
        console.log(tree);
        expect(tree).toMatchSnapshot();
    })
    it('should return log when type is log and allowWrite is false', () => {
        const component = renderer.create(
            <Work type="log" allowWrite={false}/>,
        );
        let tree = component.toJSON();
        console.log(tree);
        expect(tree).toMatchSnapshot();
    })
    it('should return mark if user is selected', () => {
        const component = renderer.create(
            <Work type="user" allowWrite={false}/>,
        );
        let tree = component.toJSON();
        console.log(tree);
        expect(tree).toMatchSnapshot();
    })
})