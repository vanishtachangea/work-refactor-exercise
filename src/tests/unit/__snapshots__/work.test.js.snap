exports[`Refactored Version of Work should return log when type is log and allowWrite is false 1`] = `
<div>
  <div>
    <div>
      Obj Saved in Remote DB
    </div>
  </div>
</div>
`;

exports[`Refactored Version of Work should return log when type is log and allowWrite is true 1`] = `
<div>
  test default
</div>
`;

exports[`Refactored Version of Work should return mark if user is selected 1`] = `
<div>
  <div />
   
</div>
`;

exports[`Refactored Version of Work should return result from recordClient.post. when type is record 1`] = `
<div>
  <div>
    <div>
      <p>
        Leanne Graham
      </p>
    </div>
    <div>
      <p>
        Sincere@april.biz
      </p>
    </div>
    <div>
      <p>
        Ervin Howell
      </p>
    </div>
    <div>
      <p>
        Shanna@melissa.tv
      </p>
    </div>
  </div>
</div>
`;

exports[`Refactored Version of Work should return result from userClient.get, when type is passed 1`] = `
<div>
  <div />
   
</div>
`;
